tests/integ/modin/frame/test_merge.py

```diff
@@ -513,18 +461,23 @@ def test_merge_on_index_multiindex_equal_labels(left_df, right_df, how, sort):
 
 
 def test_merge_left_index_right_index_single_to_multi(left_df, right_df, how, sort):
-    if how == "outer" and sort is False:
-        pytest.xfail(
-            "SNOW-1321662 - pandas 2.2.1 update fails when merge is outer and sort is False"
-        )
     right_df = right_df.rename(columns={"A": "left_i"}).set_index(
         "left_i", append=True
     )  # index columns ['right_i', 'left_i']
     if how in ("inner", "right"):
-        with SqlCounter(query_count=3, join_count=1):
-            _verify_merge(
-                left_df, right_df, how=how, left_index=True, right_index=True, sort=sort
-            )
+        if how == "inner" and sort is False:
+            # Bug in pandas. it retains right order instead of left order.
```

Comment:
could you link to the pandas bug here?

Reply:
Didn't find this bug reported in issues. I think mult-index dataframes are not really that popular so lot of bugs around these are not filed.

Reply:
ditto about pandas bug